src/client/pages/AVGDetail/AVGStatusLines.tsx

```diff
@@ -0,0 +1,157 @@
+import { AVGRequest } from '../../../server/services/avg/types';
+import { StatusLineItem } from '../../../universal/types';
+import { StatusLine } from '../../components';
+import styles from './AVGDetail.module.scss';
+
+function useAvgStatusLines(request: AVGRequest): StatusLineItem[] {
+  const isDone = !!request.datumAfhandeling;
+  const extraInfoActive = !!request.opschortenGestartOp;
+  const inProgressActive = !!request.datumInBehandeling;
+  const noProgress = !inProgressActive && !extraInfoActive && !isDone;
+
+  const inProgressDate = new Date(request.datumInBehandeling);
+  const extraInfoDate = new Date(request.opschortenGestartOp);
+
+  const lineItems = [
+    {
+      id: 'item-1',
+      status: 'Ontvangen',
+      datePublished: request.ontvangstDatum,
+      description: '',
+      documents: [],
+      isActive: noProgress,
+      isChecked: true,
+    },
+  ];
+
+  if (noProgress) {
+    lineItems.push({
+      id: 'item-3',
+      status: 'In behandeling',
+      datePublished: request.datumInBehandeling || '',
+      description: '',
+      documents: [],
+      isActive: false,
+      isChecked: true,
+    });
+
+    lineItems.push({
+      id: 'last-item',
+      status: 'Afgehandeld',
+      datePublished: request.datumAfhandeling || '',
+      description: '',
+      documents: [],
+      isActive: false,
+      isChecked: false,
+    });
+
+    return lineItems;
+  }
+
+  if (!extraInfoActive && inProgressActive) {

```

Comment:
Je zou deze nested ` if else ` ook in een switch kunnen zetten:

```
swtich (true) {

  case !extraInfoActive && inProgressActive:
    ...
  break;
  case extraInfoActive && !inProgressActive:
    ...
  break;
  case extraInfoActive && inProgressActive && inProgressDate <= extraInfoDate:
    ...
  break;
  case extraInfoActive && inProgressActive && !(inProgressDate <= extraInfoDate):
    ...
  break;
}

```

Misschien wat leesbaarder?